pennylane/capture/base_interpreter.py

```diff
@@ -0,0 +1,415 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule defines a strategy structure for defining custom plxpr interpreters
+"""
+
+import copy
+from functools import partial, wraps
+from typing import Callable
+
+import jax
+
+import pennylane as qml
+
+from .flatfn import FlatFn
+from .primitives import (
+    AbstractMeasurement,
+    AbstractOperator,
+    adjoint_transform_prim,
+    cond_prim,
+    ctrl_transform_prim,
+    for_loop_prim,
+    grad_prim,
+    jacobian_prim,
+    qnode_prim,
+    while_loop_prim,
+)
+
+
+def jaxpr_to_jaxpr(
+    interpreter: "PlxprInterpreter", jaxpr: "jax.core.Jaxpr", consts, *args
+) -> "jax.core.Jaxpr":
+    """A convenience uility for converting jaxpr to a new jaxpr via an interpreter."""
+
+    def f(*inner_args):
+        return interpreter.eval(jaxpr, consts, *inner_args)
```

Comment:
Maybe for readability/simplicity?
```suggestion
    f = partial(interpreter.eval, jaxpr, consts)
```